File: app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  include ChatHelper
  def subscribed
    stream_from channel_name
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  private

  def channel_name
    sender = fetch_user(params[:sender_type], params[:sender_id])
    receiver = fetch_user(params[:receiver_type], params[:receiver_id])
    generate_channel_name(sender, receiver)
  end

  def fetch_user(user_type, user_id)
    return AdminUser.find_by(id: user_id) if user_type == 'admin'

    User.find_by(id: user_id)
  end
end

File: app/javascript/channels/chat_channel.js
import consumer from "./consumer"

var cableData = document.getElementById('cabledata');
var senderId = cableData.dataset.userId;
var receiverId = cableData.dataset.adminUserId

consumer.subscriptions.create({ channel: "ChatChannel",
                                receiver_id: receiverId,
                                receiver_type: 'admin',
                                sender_id: senderId,
                                sender_type: 'user' }, {
  connected() {
    console.log()
  },

  disconnected() {
  },

  received(data) {
    console.log(data);
    var tableBody = document.querySelector("#table-container tbody");
    var newRow = document.createElement("tr");
    newRow.innerHTML = `
      <td>${data[1]}</td>
      <td>${data[0].message}</td>
    `;
    tableBody.appendChild(newRow);
  },
});